x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/job/RollupJobTask.java

```diff
@@ -153,59 +153,60 @@ protected void onAbort() {
     private final SchedulerEngine schedulerEngine;
     private final ThreadPool threadPool;
     private final Client client;
+    private RollupJobStatus currentState;
     private RollupIndexer indexer;
 
-    RollupJobTask(long id, String type, String action, TaskId parentTask, RollupJob job,
+    RollupJobTask(long id, String type, String action, TaskId parentTask, RollupJob job, RollupJobStatus state,
                   Client client, SchedulerEngine schedulerEngine, ThreadPool threadPool, Map<String, String> headers) {
         super(id, type, action, RollupField.NAME + "_" + job.getConfig().getId(), parentTask, headers);
         this.job = job;
         this.schedulerEngine = schedulerEngine;
         this.threadPool = threadPool;
         this.client = client;
+        this.currentState = state;
```

Comment:
I think it might makes sense to split this into `initalPosition` and `initialState`.